--- make.orig	2016-12-13 23:21:24.000000000 +0800
+++ make	2016-12-19 13:13:02.556939026 +0800
@@ -23,7 +23,10 @@
 CC =	$CC
 CFLAGS = $CFLAGS
 CPP =	$CPP
-LINK =	$LINK
+#LINK =	$LINK
+
+CXXFLAGS = -g -O0 -std=c++11 -Wall
+LINK = g++
 
 END
 
@@ -425,10 +428,16 @@
 
         ngx_src=`echo $ngx_src | sed -e "s/\//$ngx_regex_dirsep/g"`
 
+        ext=${ngx_src##*.}
+        ngx_cxx=$ngx_cc
+        if [ $ext = "cpp" ]; then
+                   ngx_cxx="$ngx_cc \$(CXXFLAGS)"
+        fi
+
         cat << END                                            >> $NGX_MAKEFILE
 
 $ngx_obj:	\$(ADDON_DEPS)$ngx_cont$ngx_src
-	$ngx_cc$ngx_tab$ngx_objout$ngx_obj$ngx_tab$ngx_src$NGX_AUX
+	$ngx_cxx$ngx_tab$ngx_objout$ngx_obj$ngx_tab$ngx_src$NGX_AUX
 
 END
      done
@@ -660,10 +669,16 @@
 END
         else
 
+            ext=${ngx_source##*.}
+            ngx_cxx=$ngx_cc
+            if [ $ext = "cpp" ]; then
+                       ngx_cxx="$ngx_cc \$(CXXFLAGS)"
+            fi
+
             cat << END                                        >> $NGX_MAKEFILE
 
 $ngx_obj:	\$(ADDON_DEPS)$ngx_cont$ngx_src
-	$ngx_cc$ngx_tab$ngx_objout$ngx_obj$ngx_tab$ngx_src$NGX_AUX
+	$ngx_cxx$ngx_tab$ngx_objout$ngx_obj$ngx_tab$ngx_src$NGX_AUX
 
 END
 
